### PR TITLE
#2 book dto

### PR DESCRIPTION
**Pull request by yanna-stepanova**

### CODE REVIEW

File: src/main/java/com/yanna/stepanova/model/Book.java [hunk 1]
```diff
@@ -9,8 +9,8 @@
 import java.math.BigDecimal;
 import lombok.Data;
 
-@Entity
 @Data
```
Comment (koval-kn) [inline src/main/java/com/yanna/stepanova/model/Book.java]: dont use data with entity as it cause bugs, replace with getter/setter annotations. fix in all models

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -6,3 +6,5 @@ spring.datasource.username=yana
 spring.datasource.password=111-Syna
 spring.jpa.hibernate.ddl-auto=create-drop
 spring.jpa.show-sql=true
+spring.jpa.open-in-view=false
+#spring.jackson.deserialization.fail-on-unknown-properties=true
```
Comment (koval-kn) [inline src/main/resources/application.properties]:
```suggestion
```

File: src/main/java/com/yanna/stepanova/controller/BookController.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/books")
```
Comment (koval-kn) [inline src/main/java/com/yanna/stepanova/controller/BookController.java]: GET: /api/books (Retrieve book catalog)
or put `api/` in application.properties servlet context path property
```suggestion
@RequestMapping("/api/books")
```

**Review by koval-kn — CHANGES_REQUESTED** [review]

File: src/main/java/com/yanna/stepanova/controller/BookController.java (hunk 3 above)
Comment (yanna-stepanova) [reply src/main/java/com/yanna/stepanova/controller/BookController.java]: Can't understand what I should do exactly
There're my screens from Postman:
https://drive.google.com/file/d/1Z5y5h22_gleTP3DLAza4hv_DbhIVT1fw/view?usp=sharing
https://drive.google.com/file/d/1iChV0GI7wAfTxBRCUipPcRJJeld9D3x6/view?usp=sharing

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -6,3 +6,5 @@ spring.datasource.username=yana
 spring.datasource.password=111-Syna
 spring.jpa.hibernate.ddl-auto=create-drop
 spring.jpa.show-sql=true
+spring.jpa.open-in-view=false
+server.servlet.context-path=/onlineBookStore
```
Comment (sarakhmen) [inline src/main/resources/application.properties]:
```suggestion
server.servlet.context-path=/api
```

**Review by sarakhmen — APPROVED** [review]
lgtm!

File: src/main/java/com/yanna/stepanova/dto/BookDto.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package com.yanna.stepanova.dto;
+
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class BookDto {
```
Comment (Elena-Bruyako) [inline src/main/java/com/yanna/stepanova/dto/BookDto.java]: You could use record. It's perfect for Dto

File: src/main/java/com/yanna/stepanova/dto/CreateBookRequestDto.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package com.yanna.stepanova.dto;
+
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class CreateBookRequestDto {
```
Comment (Elena-Bruyako) [inline src/main/java/com/yanna/stepanova/dto/CreateBookRequestDto.java]: The same

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Left comments to improve your code.

**Review by okuzan — APPROVED** [review]

**Pull request merged**